packages/api/tests/models/test-providers-model.js

```diff
@@ -139,3 +167,36 @@ test('Providers.update() throws a ValidationError if an invalid host is used', a
     t.is(err.name, 'ValidationError');
   }
 });
+
+test('Providers.update() encrypts the credentials using KMS', async (t) => {
+  const providersModel = new Provider();
+
+  const provider = fakeProviderFactory({
+    username: 'my-username-1',
+    password: 'my-password-1'
+  });
+
+  await providersModel.create(provider);
```

Comment:
Would it be better for this initial provider record to be written with plaintext credentials just to prove the process of updating plaintext -> encrypted? It ultimately doesn't matter much I think because you're updating the credentials that get encrypted and verifying that the updated values come back

Reply:
Yeah, whatever values are there are going to be overwritten, and this test only cares about what those new values are. Incidentally, the providers model doesn't provide a way to store plaintext values anyway. They're always encrypted.